hatice-ecevit-2016400138/assignment1/Anagram.java

```diff
@@ -0,0 +1,69 @@
+ import java.util.Arrays;
+ import java.util.HashMap;
+public class Anagram {
+
+    public static void main(String[] args) {
+        System.out.println(areAnagrams("apple","appll",true));
+        System.out.println(areAnagrams("listen","Silent",false));
+        System.out.println(areAnagramSentences("hi there hello","iH tHree lloeh",false));
+    }
+    public static boolean areAnagrams(String s1, String s2, boolean caseSensitive){
+        if(!caseSensitive){
```

Comment:
Style:
A few suggestions from the Java style guide:

- add a space after keywords like `if` and `for`.
- add a space before `{`
- add spaces around `=` and other operators
- also add `{` and `}`, even if there is a single statement within the `if`-, `for`-, etc statement
- use indentation consistently (Google's style guide recommends 2-space indentation)
- add a blank line to separate methods

And example:

```java
    public static void main(String[] args) {
        ...
    }

    public static boolean areAnagrams(String s1, String s2, boolean caseSensitive) {
        if (!caseSensitive) {
            s1 = s1.toLowerCase();
            s2 = s2.toLowerCase();
        }
        HashMap<Character,Integer> letterMap = new HashMap<Character, Integer>();
        if (s1.length() != s2.length()) {
            return false;
        } else {
            ...
        }
```

Some of these rules just make the code easier to read (the extra spaces, for example), less error prone (since you always have `{` and `}`, you do not accidentally add a statement that looks like it is within the `then`-branch, but actually it is not), other are more arbitrary (no strong rationale 2-space instead of 4-space indentation) but help with consistency (when you work together with someone else on some code, if everyone uses a different indentation, it means that every time you make an edit you either have to change it back and forth, or the indentation becomes inconsistent).

Reply:
Here is a link to the Google Java style guide:
https://google.github.io/styleguide/javaguide.html

It is rather massive, so not a very easy read, but the way we usually learn it at Google is via code reviews.